plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/IcebergQueryRunner.java

```diff
@@ -46,64 +51,70 @@ public static DistributedQueryRunner createIcebergQueryRunner()
             Map<String, String> extraProperties,
             Map<String, String> connectorProperties,
-            Iterable<TpchTable<?>> tables,
-            Optional<File> metastoreDirectory)
+            Optional<File> metastoreDirectory,
+            Consumer<QueryRunner> schemaInitializer)
             throws Exception
     {
         Session session = testSessionBuilder()
                 .setCatalog(ICEBERG_CATALOG)
-                .setSchema("tpch")
+                .setSchema(SCHEMA)
                 .build();
 
         DistributedQueryRunner queryRunner = DistributedQueryRunner.builder(session)
                 .setExtraProperties(extraProperties)
                 .build();
 
         queryRunner.installPlugin(new TpchPlugin());
-        queryRunner.createCatalog("tpch", "tpch");
-
-        Path dataDir = metastoreDirectory.map(File::toPath).orElseGet(() -> queryRunner.getCoordinator().getBaseDataDir().resolve("iceberg_data"));
+        queryRunner.createCatalog("tpch", SCHEMA);
 
         queryRunner.installPlugin(new IcebergPlugin());
         connectorProperties = new HashMap<>(ImmutableMap.copyOf(connectorProperties));
-        connectorProperties.putIfAbsent("iceberg.catalog.type", "TESTING_FILE_METASTORE");
-        connectorProperties.putIfAbsent("hive.metastore.catalog.dir", dataDir.toString());
+        if (!connectorProperties.containsKey("iceberg.catalog.type")) {
+            Path dataDir = metastoreDirectory.map(File::toPath).orElseGet(() -> queryRunner.getCoordinator().getBaseDataDir().resolve("iceberg_data"));
+            connectorProperties.putIfAbsent("iceberg.catalog.type", "TESTING_FILE_METASTORE");
+            connectorProperties.putIfAbsent("hive.metastore.catalog.dir", dataDir.toString());
+        }
         queryRunner.createCatalog(ICEBERG_CATALOG, "iceberg", connectorProperties);
-
-        queryRunner.execute("CREATE SCHEMA tpch");
-
-        copyTpchTables(queryRunner, "tpch", TINY_SCHEMA_NAME, session, tables);
-
+        schemaInitializer.accept(queryRunner);
         return queryRunner;
     }
 
+    public static void initializeSchema(QueryRunner queryRunner)
+    {
+        queryRunner.execute("CREATE SCHEMA " + SCHEMA);
+        copyTpchTables(queryRunner, "tpch", TINY_SCHEMA_NAME, queryRunner.getDefaultSession(), ImmutableList.of(NATION, REGION));
+    }
+
     public static void main(String[] args)
             throws Exception
     {
         Logging.initialize();
         Map<String, String> properties = ImmutableMap.of("http-server.http.port", "8080");
         DistributedQueryRunner queryRunner = null;
         try {
-            queryRunner = createIcebergQueryRunner(properties, Map.of(), TpchTable.getTables());
+            queryRunner = createIcebergQueryRunner(properties, Map.of(), Optional.empty(), runner -> {
+                initializeSchema(runner);
+                copyTpchTables(runner, "tpch", TINY_SCHEMA_NAME, runner.getDefaultSession(), TpchTable.getTables());
```

Comment:
this executes `copyTpchTables` twice
rather than that, just call `"CREATE SCHEMA " + runner.getDefaultSession().getSchema().orElseThrow()`

Reply:
Applied